sqlglot/dialects/snowflake.py

```diff
@@ -1,11 +1,105 @@
-from sqlglot.dialects.dialect import Dialect
+from sqlglot import exp
+from sqlglot.dialects.dialect import Dialect, format_time_lambda, rename_func
+from sqlglot.expressions import Literal
+from sqlglot.helper import list_get
+from sqlglot.parser import Parser
 from sqlglot.tokens import Tokenizer, TokenType
+from sqlglot.generator import Generator
+
+
+def _check_int(s):
+    if s[0] in ("-", "+"):
+        return s[1:].isdigit()
+    return s.isdigit()
+
+
+# from https://docs.snowflake.com/en/sql-reference/functions/to_timestamp.html
+def _snowflake_to_timestamp(args):
+    if len(args) == 2:
+        first_arg, second_arg = args
+        if first_arg.is_string:
+            # case: <string_expr> [ , <format> ]
+            return format_time_lambda(exp.StrToTime, "snowflake")(args)
+
+        # case: <numeric_expr> [ , <scale> ]
+        if second_arg.this not in ["0", "3", "9"]:
+            raise ValueError(
+                f"Scale for snowflake numeric timestamp is {second_arg}, but should be 0, 3, or 9"
+            )
+
+        timestamp = int(first_arg.name)
+        scale = int(second_arg.name)
+        args[:] = args[:1]
+        return exp.UnixToTime(this=str(int(int(timestamp) / (10 ** int(scale)))))
```

Comment:
remove the ints from timestamp and scale because it's not necessary,

you also don't need int(x / y)  you can just do timestamp // (10 ** scale)

Reply:
fixed